packages/dataviews/src/field-types/datetime.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * WordPress dependencies
+ */
+import { BaseControl } from '@wordpress/components';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	SortDirection,
+	ValidationContext,
+	DataFormControlProps,
+} from '../types';
+
+function sort( a: any, b: any, direction: SortDirection ) {
+	const timeA = new Date( a ).getTime();
+	const timeB = new Date( b ).getTime();
+
+	return direction === 'asc' ? timeA - timeB : timeB - timeA;
+}
+
+function isValid( value: any, context?: ValidationContext ) {
+	if ( context?.elements ) {
+		const validValues = context?.elements.map( ( f ) => f.value );
+		if ( ! validValues.includes( value ) ) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+function Edit< Item >( {
+	data,
+	field,
+	onChange,
+}: DataFormControlProps< Item > ) {
+	const { id, label } = field;
+	const value = field.getValue( { item: data } );
+
+	const onChangeDate = useCallback(
+		( event: any ) => {
+			const getNewValue = ( prevValue: string, newDate: string ) => {
+				const prevTime =
+					prevValue?.split( 'T' )[ 1 ] ||
+					new Date().toISOString().split( 'T' )[ 1 ];
+
+				// TODO: WordPress uses ISO8601,
+				// we should allow the consumer to declare the date format.
+				return `${ newDate }T${ prevTime }`;
```

Comment:
The date will be serialized in ISO8601. This is what WordPress uses and it's a widespread format. Is it reasonable to ask consumers to use this format in fields of type `date`? Or should we allow consumers to specify the format in which the data comes (and that we should respect in serialization)?

Reply:
I believe `format` is something important and should be something passed by the consumer. We can have a reasonable default like the one you suggest.

Reply:
By using the `TimePicker` component this is no longer relevant / under our control. It'll take any date input and will output it as ISO8601. The consumer may adapt it to a different format should they need.